feature/experimental/bgp/ate_tests/bgp_remove_private_as/bgp_remove_private_as_test.go

```diff
@@ -0,0 +1,353 @@
+	case "SET":
+		// TODO : SET mode is not working
+		// https://github.com/openconfig/featureprofiles/issues/1659
+		bgpNeti1.BGP().AddASPathSegment(asSeg...).WithTypeSET()
+	default:
+		bgpNeti1.BGP().AddASPathSegment(asSeg...)
+	}
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	return topo
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+	peerGrp    string
+}
+
+// verifyBGPAsPath is to Validate AS Path attribute using bgp rib telemetry on ATE.
+func verifyBGPAsPath(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice, asSeg []uint32, removeASPath bool) {
+	at := gnmi.OC()
+	rib := at.NetworkInstance(ateDst.Name).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "0").Bgp().Rib()
+	prefixPath := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
+		NeighborAny().AdjRibInPre().RouteAny().WithPathId(0).Prefix()
+
+	gnmi.WatchAll(t, ate, prefixPath.State(), time.Minute, func(v *ygnmi.Value[string]) bool {
+		_, present := v.Val()
+		return present
+	}).Await(t)
+
+	var wantAsSeg = []uint32{dutAS, ateAS1}
+	if !removeASPath {
+		wantAsSeg = append(wantAsSeg, asSeg...)
+	} else {
+		for _, as := range asSeg {
+			if as < 64512 {
+				wantAsSeg = append(wantAsSeg, as)
+			}
+		}
+	}
+	var gotAsSeg []uint32
+	_, ok := gnmi.WatchAll(t, ate, rib.AttrSetAny().AsSegmentAny().State(), 1*time.Minute, func(v *ygnmi.Value[*oc.NetworkInstance_Protocol_Bgp_Rib_AttrSet_AsSegment]) bool {
+		val, present := v.Val()
+		gotAsSeg = val.Member
+		return present && cmp.Diff(val.Member, wantAsSeg) == ""
+	}).Await(t)
+	if !ok {
+		t.Errorf("Obtained AS path on ATE is not as expected, gotAsSeg %v, wantAsSeg %v", gotAsSeg, wantAsSeg)
+	}
+}
+
+// TestRemovePrivateAS is to Validate that private AS numbers are stripped
+// before advertisement to the eBGP neighbor.
+func TestRemovePrivateAS(t *testing.T) {
+	t.Logf("Start DUT config load.")
+	dut := ondatra.DUT(t, "dut")
+
+	t.Run("Configure DUT interfaces", func(t *testing.T) {
+		t.Logf("Start DUT interface Config.")
+		configureDUT(t, dut)
+	})
+
+	t.Run("Configure DEFAULT network instance.", func(t *testing.T) {
+		t.Log("Configure Network Instance type to DEFAULT.")
+		dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
+		gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	})
+
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	t.Run("Configure BGP Neighbors.", func(t *testing.T) {
+		t.Logf("Start DUT BGP Config.")
+		gnmi.Delete(t, dut, dutConfPath.Config())
+		dutConf := bgpCreateNbr(dutAS, ateAS1, dut)
+		gnmi.Replace(t, dut, dutConfPath.Config(), dutConf)
+		fptest.LogQuery(t, "DUT BGP Config", dutConfPath.Config(), gnmi.GetConfig(t, dut, dutConfPath.Config()))
+	})
+
+	cases := []struct {
```

Comment:
Add some 4-byte private ASNs to the mix? (>=4,200,000,000)

Reply:
SET method is TODO, we can 4byte ASN to SET mode.